server/apps/tasks.py

```diff
@@ -228,10 +231,14 @@ def on_update(self, updates, original):
         old_stage_id = original.get('task', {}).get('stage', '')
         new_user_id = updates.get('task', {}).get('user', '')
         if new_stage_id and new_stage_id != old_stage_id:
+            updates[ITEM_OPERATION] = ITEM_SEND
             send_to(doc=original, update=updates, desk_id=None, stage_id=new_stage_id, user_id=new_user_id)
-        update_version(updates, original)
+        resolve_document_version(updates, ARCHIVE, 'PATCH', original)
 
     def on_updated(self, updates, original):
+        updated = copy(original)
+        updated.update(updates)
+        insert_versioning_documents(ARCHIVE, updated)
```

Comment:
If the article is assigned to a new desk then the `updated` dict is inserted twice. Please refer 
https://github.com/superdesk/superdesk/blob/master/server/apps/tasks.py#L249-L252